docs/docs/for-developers/architecture.md

```diff
@@ -0,0 +1,105 @@
+---
+id: architecture
+title: Architecture Overview
+sidebar_label: Architecture
+---
+
+## Architecture
+
+Onivim 2 is a [Revery](https://www.github.com/revery-ui/revery) application built with [ReasonML](https://reasonml.github.io).
+
+Even though it is new technology, if you're coming from a web background - it should be easy to get started! The syntax of ReasonML is similar to JavaScript,
+and the core ideas are very similar to [React](https://reactjs.org), [Redux](https://redux.js.org/), and the [Elm Architecture](https://guide.elm-lang.org/architecture/) - just used in a way that lets us compile to native code.
+
+We want to have a functional lean to our codebase... and [ReasonML](https://reasonml.github.io) / [OCaml](https://ocaml.org) are the perfect fit for that.
+[OCaml](https://ocaml.org) is highly optimized - even down to its garbage collector implementation - for building functional applications in a performant way.
+
+### Key concepts
+
+- __Action__: A payload of information to describe a state change. Just like an [action](https://redux.js.org/basics/actions) in redux.
+- __State__ and __Model__: used interchangeably to describe the application state. 
+- __Reducer__: A function of (`state`, `action`) that returns a new `state`.
+- __Updater__: A function of (`state`, `action`) that returns a tuple of (`state`, `effects`). Similar to a reducer, but also handles side-effects. Inspired by the [Elm Architecture](https://guide.elm-lang.org/architecture/)
+- __Store Connector__: Loosely akin to a middleware in React.
```

Comment:
I think it's suppose to be `Loosely akin to a middleware in redux` instead of  `Loosely akin to a middleware in React`.

Reply:
Nice catch, thanks @KevinSjoberg !